tree/dataframe/src/RLoopManager.cxx

```diff
@@ -474,9 +541,23 @@ void RLoopManager::RunTreeProcessorMT()
 /// Run event loop over one or multiple ROOT files, in sequence.
 void RLoopManager::RunTreeReader()
 {
+   if (fStartEntry == fEndEntry) // empty range => no work needed
+      return;
+
    TTreeReader r(fTree.get(), fTree->GetEntryList());
+
    if (0 == fTree->GetEntriesFast())
       return;
+
+   // if `GetEntries` is not called, in case of chains, `SetEntriesRange` might not properly indicate that a
+   // start entry is too large, e.g. when start=last entry
+   // the `GetEntries` call can be removed once `https://github.com/root-project/root/issues/10774` is resolved
+   if (fStartEntry >= r.GetEntries(true))
+      throw std::logic_error(std::string("A range of entries was passed in the creation of the RDataFrame, ") +
+                             "but the starting entry is larger than the total number of entries (" +
+                             r.GetEntries(true) + ") in the dataset.");
```

Comment:
saves some work (no call to `LoadTree`) as we know the true number of entries has been loaded at this point:

```suggestion
                             r.GetEntries() + ") in the dataset.");
```